File: cpp/water.cpp
// water.cpp
#include "metaclass.hpp"
#include <iostream>

class Water : public Metaclass {
public:
    void displayAttributes() const override {
        std::cout << "Water Attributes:\n"
                  << "State: Liquid\n"
                  << "Temperature: 0-100C\n"
                  << "Flow: Yes\n"
                  << "Evaporation: Yes\n"
                  << "Freezing: Yes\n";
    }
};

File: rust/water.rs
use crate::metaclass::Metaclass;

pub struct Water;

impl Metaclass for Water {
    fn display_attributes(&self) {
        println!("Water Attributes:");
        println!("State: Liquid");
        println!("Temperature: 0-100C");
        println!("Flow: Yes");
        println!("Evaporation: Yes");
        println!("Freezing: Yes");
    }
}

